mne/annotations.py

```diff
@@ -286,6 +337,26 @@ def orig_time(self):
         """The time base of the Annotations."""
         return self._orig_time
 
+    @property
+    def details(self):
+        """The details of the Annotations."""
+        return self._details
+
+    @details.setter
+    def details(self, details):
+        self._details = _validate_details(details, len(self.onset))
+
+    @property
+    def details_columns(self) -> set[str]:
+        """The set containing all the keys in all details dicts."""
+        return {k for d in self.details if d is not None for k in d.keys()}
+
+    @property
+    def details_data_frame(self):
+        """The details of the Annotations as a DataFrame."""
+        pd = _check_pandas_installed(strict=True)
+        return pd.DataFrame([d if d is not None else {} for d in self.details])
```

Comment:
I don't get how you can have None in the list

Reply:
if you have say 3 annotation spans, and second one is `BAD_EDGE` then that one probably won't have any extra metadata attached to it.  So the list would be `[dict(...), None, dict(...)]`

Reply:
hum... my first reaction is then to have an empty dict unless there is an implicit assumption that the keys in each dictionary should be the same for all annotations unless it's None.

here the design is via an array of structures (list of dicts) while onset, duration and description are structures of arrays (dict-like of arrays). The latter forces all annotations to have the same schema. That would be my inclination to force the same schema unless it's deemed too restrictive.

my 2c

Reply:
I modified such that:
- internally, extras is a list of dicts. This simplifies the code and avoids the None case
- when instantiating, the users still have the possibility to pass a `list[dict | None]` for simplicity